fluent/fluent.go

```diff
@@ -180,25 +213,21 @@ func (f *Fluent) EncodeData(tag string, tm time.Time, message interface{}) (data
 
 // Close closes the connection.
 func (f *Fluent) Close() (err error) {
-	if len(f.pending) > 0 {
-		err = f.send()
+	f.cancel()
+	for err = range f.result {
+		f.close()
+		break
 	}
-	f.close()
 	return
 }
 
 // close closes the connection.
 func (f *Fluent) close() (err error) {
```

Comment:
How about removing return value since this function is private and always returns nil?